Modules/SubModule_Population/xASL_adm_GzipAllFiles.m

```diff
@@ -39,47 +43,65 @@ function xASL_adm_GzipAllFiles(ROOT, bFolder, bUseLinux)
         PathToSearch = xASL_adm_UnixPath(ROOT);
         system(['for i in `find ' PathToSearch '/* | grep -E \.nii$`; do gzip -1 -f -q -v "$i"; done'], '-echo');
         return;
+    else
+        %% 2) Otherwise use the multithreaded SuperGzip for Windows
+        PathList = xASL_adm_GetFileList(ROOT, '^.*\.(nii)$', 'FPListRec', [0 Inf], false);
+        fprintf('\n%s\n',['G-zZzZipping ' num2str(length(PathList)) ' files']);
+        n_THREADS = min([str2num(getenv('NUMBER_OF_PROCESSORS')) length(PathList)]);
+        if ~isempty(EXTERNAL)
+            PathToSuperGzip = fullfile(EXTERNAL, 'SuperGZip', 'SuperGZip_Windows.exe');
+            command = [PathToSuperGzip ' -p 0 -n ' num2str(n_THREADS) ' -v 1 ' ROOT ' *.nii'];
+            exit_code = system(command);
+        else
+            exit_code = 1;
+        end
+        if exit_code == 0
+            fprintf('Gzipping Niftis Successful...\n')
+        else
+            warning('An error occurred while using SuperGzip')
+        end
     end
 
     %% ----------------------------------------------------
     %% 2) Slower Matlab version
     if bFolder
```

Comment:
Also, is the `bFolder` option still respected? This would zip a folder completely, and this is actually not what we want (which is why it is an option). We want each `.nii` to be replaced by `.nii.gz`.

Reply:
Reverted 👍